### PR TITLE
feat(aws-cur): add new property `enableDefaultUniqueReportName` that creates a unique cost report name by default

### PR DESCRIPTION
**Pull request by go-to-k**
### Issue # (if applicable)

Closes #<issue number here>.

### Reason for this change

Cost report names must be unique, but the current construct default names do not ensure uniqueness. (Because the default name is `default-cur`.)

### Description of changes

Added `enableDefaultUniqueReportName` property.

The construct generates a default name using `Names.uniqueResourceName` if the `enableDefaultUniqueReportName` is set to true. This flag is false by default, in which case the default report name remains `default-cur`.

Also, the necessary validations have been included.

### Description of how you validated changes

Both unit and integ tests.

### Checklist

- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/open-constructs/aws-cdk-library/blob/main/CONTRIBUTING.md)

---
_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: src/aws-cur/cost-report.ts [hunk 1]
```diff
@@ -150,6 +174,8 @@ export class CostReport extends Construct {
       additionalSchemaElements: ['RESOURCES'],
     });
     reportDefinition.node.addDependency(this.reportBucket.policy!);
+
+    this.costReportName = reportDefinition.ref;
```
Comment (go-to-k) [inline src/aws-cur/cost-report.ts:178]: To obtain an auto-generated name.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cur-reportdefinition.html#aws-resource-cur-reportdefinition-return-values

File: src/aws-cur/cost-report.ts [hunk 2]
```diff
@@ -137,11 +143,29 @@ export class CostReport extends Construct {
...
+        throw new Error(
+          `'costReportName' must be between 1 and 256 characters long, got: ${props.costReportName.length}`,
+        );
+      }
+      if (!/^[0-9A-Za-z!\-_.*'()]+$/.test(props.costReportName)) {
+        throw new Error(
+          `'costReportName' must only contain alphanumeric characters and the following special characters: !-_.*'(), got: '${props.costReportName}'`,
+        );
+      }
+    }
```
Comment (go-to-k) [inline src/aws-cur/cost-report.ts:157]: ref: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cur-reportdefinition.html#cfn-cur-reportdefinition-reportname

**Comment by go-to-k** [comment]
Waiting for the [PR](https://github.com/open-constructs/aws-cdk-library/pull/50) to be merged.

**Comment by go-to-k** [comment]
@hoegertn Updated doc to ensure successful build. Can we merge it?

**Comment by hoegertn** [comment]
I think the change makes sense. My concern is that this breaks users currently using this construct as it changes the name if the y did not set one.

Any ideas how to prevent this?

**Comment by go-to-k** [comment]
@hoegertn 
> I think the change makes sense. My concern is that this breaks users currently using this construct as it changes the name if the y did not set one.
> Any ideas how to prevent this?

Usually in such cases, there is a way to add a new property and deprecate the old one. However, if we want to change the behaviour of the default values, as in this case, we cannot use that approach.
Instead, feature flags are often used, but we need to come up with a new approach similar to the feature flags in the aws-cdk-lib.

If we still want to prevent breaking change at this point, I considered adding a property like a feature flag. See the commit: https://github.com/open-constructs/aws-cdk-library/pull/48/commits/fa8701ee9390aafca6e92ff84d0cbc97cc586e5b

I will change the title and description of this PR accordingly for now.

**Review by hoegertn — APPROVED** [review]

**Pull request merged**